src/Symfony/Component/VarDumper/Caster/RedisCaster.php

```diff
@@ -63,6 +77,71 @@ public static function castRedisArray(\RedisArray $c, array $a, Stub $stub, $isN
         return $a + array(
             $prefix.'hosts' => $c->_hosts(),
             $prefix.'function' => ClassStub::wrapCallable($c->_function()),
+            $prefix.'lastError' => $c->getLastError(),
+            $prefix.'options' => self::getRedisOptions($c),
+        );
+    }
+
+    public static function castRedisCluster(\RedisCluster $c, array $a, Stub $stub, $isNested)
+    {
+        $prefix = Caster::PREFIX_VIRTUAL;
+        $failover = $c->getOption(\RedisCluster::OPT_SLAVE_FAILOVER);
+
+        $a += array(
+            $prefix.'_masters' => $c->_masters(),
+            $prefix.'_redir' => $c->_redir(),
+            $prefix.'mode' => new ConstStub($c->getMode() ? 'MULTI' : 'ATOMIC', $c->getMode()),
+            $prefix.'lastError' => $c->getLastError(),
+            $prefix.'options' => self::getRedisOptions($c, array(
+                'SLAVE_FAILOVER' => isset(self::$failover[$failover]) ? new ConstStub(self::$failover[$failover], $failover) : $failover,
+            )),
         );
+
+        return $a;
+    }
+
+    /**
+     * @param \Redis|\RedisArray|\RedisCluster
+     */
+    private static function getRedisOptions($redis, array $options = array()): EnumStub
+    {
+        if (\is_array($serializer = $redis->getOption(\Redis::OPT_SERIALIZER))) {
+            foreach ($serializer as &$v) {
+                if (isset(self::$serializer[$v])) {
+                    $v = new ConstStub(self::$serializer[$v], $v);
+                }
+            }
+        } elseif (isset(self::$serializer[$serializer])) {
+            $serializer = new ConstStub(self::$serializer[$serializer], $serializer);
+        }
+
+        if (\is_array($compression = \defined('Redis::OPT_COMPRESSION') ? $redis->getOption(\Redis::OPT_COMPRESSION) : 0)) {
```

Comment:
I think keeping the assignment `$compression = \defined('Redis::OPT_COMPRESSION') ? $redis->getOption(\Redis::OPT_COMPRESSION) : 0` outside the `if (\is_array($compression))` would be more readable (and totally equivalent).

Reply:
updated thanks